### ISSUE TITLE
Unable to build the sdk (C++) in Windows

### ISSUE THREAD

**Issue by hirenpatel1207** [ouster-lidar/ouster-sdk#610]
**Describe your question**
Building the sdk from source from C++ throws error in compilation.

**Screenshots**
![Screenshot 2024-09-03 142837](https://github.com/user-attachments/assets/c0b1d575-299f-4b18-9932-0e246191a528)

**Platform (please complete the following information):**
- OS-1
- Ouster Firmware Version? \[e.g. 2.3, 2.4, ..\]
- C++
- Windows
- X64

Goal: Use the C++ sdk to build an application in Windows to capture and store the Lidar data (without using ros2).
I followed the instructions given at https://static.ouster.dev/sdk-docs/cpp/building.html . In the last step it is metioned to open https://github.com/ouster-lidar/ouster-sdk/archive/refs/tags/20240703.zip  in Visual studio and build the project. But I am getting errors linkage errors. 
Is it possible to get prebuilt binaries for Windows platform?

**Comment by Samahu** [comment]
Hi @hirenpatel1207, can you try to see if this branch helps you build on Windows https://github.com/ouster-lidar/ouster-sdk/pull/608

**Comment by matthew-lidar** [comment]
The changes in the PR above were merged into master, so if you use: https://codeload.github.com/ouster-lidar/ouster-sdk/zip/refs/heads/master it should work.

**Comment by hirenpatel1207** [comment]
@matthew-lidar I was able to build the master branch. Thanks.
However the build directly generates the example executables.
![image](https://github.com/user-attachments/assets/6b6bc28a-66a3-45cb-b1c7-d1afa2bdfbe2)
By that way, I would have to import the ouster-sdk source code into my project and then build it together, which is not a great idea for versioning and keeping sdk's seperate.
Is it possible to pre-built the sdk seperately and generate the .h, .lib and .dll s and then use them?

**Comment by matthew-lidar** [comment]
Usually what one would do is build the CMake package, then `install` it using a command like:
```
cmake --install . --config x64-Release
```

If you used Visual studio, there's probably also a way to trigger this, but I was unable to find documentation saying specifically how.

This will copy the .h .lib and .dll files to a central directory and enable other CMake projects, such as your's to `find_package` it.

Then do something like the following in your project's CMake and it will automatically find and include those files:

```cmake
project(your_project)
cmake_minimum_required(VERSION 3.10)

find_package(OusterSDK REQUIRED)

add_executable(your_project main.cpp)
target_link_libraries(your_project OusterSDK::ouster_client)
set_property(TARGET your_project PROPERTY CXX_STANDARD 14)
```

**Comment by hirenpatel1207** [comment]
Do you mean this option in VS?
![Screenshot 2024-09-09 120218](https://github.com/user-attachments/assets/35f0170f-a44c-4c44-8c74-8b9d0d6ebf31)
[install.zip](https://github.com/user-attachments/files/16928877/install.zip)

With that option, I was able to generate the ouster_client.lib, ouster_osf.lib, ouster_pcap.lib,  and ouster_viz.lib files (see the attached install.zip folder).
But there are no dlls.
Also I tried to use these header and .lib files in a QT project .pro file but I am getting the following error:
![image](https://github.com/user-attachments/assets/7943600e-eb8a-49f8-b4ff-ba3f976ef0c8)
I think something related to networking stuff is not getting generated in the install step.

**Comment by hirenpatel1207** [comment]
> Do you mean this option in VS? ![Screenshot 2024-09-09 120218](https://private-user-images.githubusercontent.com/45404771/365606744-35f0170f-a44c-4c44-8c74-8b9d0d6ebf31.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU4NzY5MjIsIm5iZiI6MTcyNTg3NjYyMiwicGF0aCI6Ii80NTQwNDc3MS8zNjU2MDY3NDQtMzVmMDE3MGYtYTQ0Yy00YzQ0LThjNzQtOGI5ZDBkNmViZjMxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA5VDEwMTAyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI1M2EwYzEyMWIxNWQ1ZTg1ZTI5ZjVkOTc4NWQ4NWZkNzcwMGY1YjFlZTM3NzFmMjI4YTdhOWNjNGUyOTdhZjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.No627EvV3tMML3UPAqSG7DuTQcwWLLp2gJQH9yVDUiA) [install.zip](https://github.com/user-attachments/files/16928877/install.zip)
> 
> With that option, I was able to generate the header files and ouster_client.lib, ouster_osf.lib, ouster_pcap.lib, and ouster_viz.lib files (see the attached install.zip folder). But there are no dlls. Also I tried to use these header and .lib files in a QT project .pro file but I am getting the following error: ![image](https://private-user-images.githubusercontent.com/45404771/365608369-7943600e-eb8a-49f8-b4ff-ba3f976ef0c8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU4NzY5MjIsIm5iZiI6MTcyNTg3NjYyMiwicGF0aCI6Ii80NTQwNDc3MS8zNjU2MDgzNjktNzk0MzYwMGUtZWI4YS00OWY4LWI0ZmYtYmEzZjk3NmVmMGM4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA5VDEwMTAyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRhMDk2MjE3YmYzZmMyMzZmMTg2ZTg2NmIwOTI1NjkyMzgyZmIyOTU1MWQ0Zjc2YmQyNmM0ZTkyM2U5NjU5ODgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.MqbiHESwxoZ8UycC1-JXthjeym1gJObaO6R1v9p4LbY) I think something related to networking stuff is not getting generated in the install step.

EDIT:
I was able to solve the problem by including "-lws2_32". 
But now getting "write access violation", I think it has something to do with the spdlog.
The program crashes at line-> ouster::sensor::init_logger("info");
Here is the minimalistic example that I am trying: 
![image](https://github.com/user-attachments/assets/a17d22d9-4b14-46a0-8dc6-23a74b5c503f)

**Comment by matthew-lidar** [comment]
Currently the SDK is only built statically, so you will not find any .dlls generated, only .libs.

As for the crash, do you have the full stacktrace?

**Comment by hirenpatel1207** [comment]
Got it. My "Hello World" sample program runs now. I had linked wrong .libs from vcpkg. Thanks for the help.

**Comment by hirenpatel1207** [comment]
Sorry for going back and forth on opening this issue as I was not sure on whether  to open a new issue or keep this one.
The build is successfully but the release library version has some problem I think.
For e.g. if i run the client_example.exe of the debug version, it works fine.
But the same client_example.exe of release version does not work.
The method sensor::poll_client(*handle); returns value "0". What does that mean?

**Comment by matthew-lidar** [comment]
On Windows you need to make sure you link debug builds of your program with debug builds of the SDK (and release builds with release builds) for things to work correctly generally. I would check to make sure that is occurring.

Timeout means that no packets were received in the timeout provided to the ``poll_client`` function. The default is 1 second.

**Comment by hirenpatel1207** [comment]
Yes, I was getting problems with my programs during the release build. Then I had gone back to check if the release version demos work or not. I have tried the demos with no modifications. I had just followed the guide line in the docs to build the package. I had checkout vcpkg 2024.04.26(git checkout 2024.04.26) so that correct libraray versions are linked.
